subsys/usb/usb_c/usbc_stack.c

```diff
@@ -113,7 +115,10 @@ int usbc_suspend(const struct device *dev)
 
 	/* Add private suspend request to fifo */
 	data->request.val = PRIV_PORT_REQUEST_SUSPEND;
-	k_fifo_put(&data->request_fifo, &data->request);
+
+	if (k_fifo_is_empty(&data->request_fifo)) {
```

Comment:
If you are preventing the subsystem from suspending, you should return a value that informs the caller
of the failure to suspend. Maybe return -EBUSY.

Reply:
It's not exactly that I am preventing the suspend. The request is inside the data object of the usb-c port instance, so only one request can exist in the same time. This function overrides the value of the request, so theoretically the request should still be valid (but it should be verified further to check for any race conditions) and overrides the previous one. Main problem can be if we use the REQUEST_NOP for state changing and other thread wants to suspend the usb-c, it needs more checks, or just check if overwritten request was not a suspend.

But the main problem here is that we have only one instance of the request object, so if the fifo is not empty, that means this object was already added to the fifo, and adding it again causes the deadlock during the iteration of the fifo by the usb-c stack.

Reply:
As we discussed, this was changed to more specific solution used entirely to prevent one thread sleep to improve the stack response time. Previous solution was a misuse of the existing logic.